core/include/array/array_sorted_read_state.h

```diff
@@ -201,17 +202,14 @@ class ArraySortedReadState {
   /** AIO counter. */
   int aio_cnt_;
 
-  /** The AIO mutex conditions (one for each buffer). */
-  pthread_cond_t aio_cond_[2];
+  /** Condition variables used in AIO. */
+  std::condition_variable aio_cv_[2];
```

Comment:
It makes the code much much easier and shorter. Right now, the id in `aio_cv_` is initialized to `0` and then it is interchanged between `1` and `0` again. In the code, the only thing we do is `id = (id + 1) %2` and then everything works with consistent naming `aio_cv_[id]`. If we use a different name, e.g., `aio_cv_0_` and  `aio_cv_0_` the code becomes a mess with `if`'s.